internal/v4/archive.go

```diff
@@ -133,13 +136,21 @@ func (h *Handler) servePostArchive(id *charm.Reference, w http.ResponseWriter, r
 		})
 	}
 
+	// Get the promulgated URL for this entity, If the user is not the
```

Comment:
Perhaps:
// Find the promulgated URL for the entity. Note: if the entity is not promulgated,
// getPromulgatedURL returns nil.

?